src/query/expression/src/evaluator.rs

```diff
@@ -1000,6 +1054,11 @@ impl<'a> Evaluator<'a> {
                     },
                     _ => unreachable!(),
                 };
+
+                if func_name == "array_fold" {
+                    return self.run_array_fold(&inner_col, &expr);
```

Comment:
`inner_col` can't be used directly in `run_array_fold` because it's an expanded column, the offset information is missing, and we can't construct a result column without offsets.